public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -153,6 +135,28 @@ export class DashboardPage extends PureComponent<Props, State> {
         this.triggerPanelsRendering.bind(this)
       );
     }
+
+    // entering edit mode
+    if (!editPanel && urlEditPanelId) {
+      this.getPanelByIdFromUrlParam(urlEditPanelId, panel => {
+        // if no edit permission show error
+        if (!dashboard.canEditPanel(panel)) {
+          this.props.notifyApp(createErrorNotification('Permission to edit panel denied'));
+          dashboard.initViewPanel(panel);
+          this.setState({
+            viewPanel: panel,
+            rememberScrollTop: this.state.scrollTop,
+          });
+        } else {
+          this.setState({ editPanel: panel });
+        }
```

Comment:
```suggestion
        if (dashboard.canEditPanel(panel)) {
          this.setState({ editPanel: panel });
          return;
        }

        this.props.notifyApp(createErrorNotification('Permission to edit panel denied'));
        this.props.updateLocation({
          query: {
            editPanel: null,
            viewPanel: panel.id,
          },
          partial: true,
          replace: true,
        });
```

After reading the comments in the this PR I would suggest doing the following. This should work as expected and also handle the browser history correctly.

Please give it a try and say what you think 🙏